### ISSUE TITLE
resample_spec dimensionless quantitiy error with nirspec MSA data

### ISSUE THREAD

**Issue by stscieisenhamer** [STScI-JWST/jwst#1005]
I believe no work has been done yet on `resample_spec` running on NIRSpec MSA data, but just to document:

```
$ pytest -rsx jwst/pipeline/tests/test_calwebb_spec3_nrs_msa.py::test_run_resample_only --runslow
============================================================================= test session starts ==============================================================================
platform darwin -- Python 3.5.3, pytest-3.1.2, py-1.4.34, pluggy-0.4.0
rootdir: /Users/eisenham/Documents/ssbdev/jwst, inifile: setup.cfg
plugins: catchlog-1.2.2, xdist-1.16.0
collected 6 items

jwst/pipeline/tests/test_calwebb_spec3_nrs_msa.py F

=================================================================================== FAILURES ===================================================================================
____________________________________________________________________________ test_run_resample_only ____________________________________________________________________________

mk_tmp_dirs = ('/var/folders/z7/hcykr_n546s_5033t9rj5d200000zj/T/tmpzsvan83z', '/var/folders/z7/hcykr_n546s_5033t9rj5d200000zj/T/tmpx19po6s9', '/var/folders/z7/hcykr_n546s_5033t9rj5d200000zj/T/tmpqi2eo0om')

    @runslow
    @require_bigdata
    def test_run_resample_only(mk_tmp_dirs):
        """Test resample step only."""
        tmp_current_path, tmp_data_path, tmp_config_path = mk_tmp_dirs

        asn_path = update_asn_basedir(
            path.join(DATAPATH, 'mos_udf_g235M_spec3_asn.json'),
            root=path.join(DATAPATH, 'level2b')
        )
        args = [
            path.join(SCRIPT_DATA_PATH, 'calwebb_spec3_default.cfg'),
            asn_path,
            '--steps.skymatch.skip=true',
            '--steps.outlier_detection.skip=true',
            '--steps.cube_build.skip=true',
            '--steps.extract_1d.skip=true',
        ]

>       Step.from_cmdline(args)

/Users/eisenham/Documents/ssbdev/jwst/jwst/pipeline/tests/test_calwebb_spec3_nrs_msa.py:69:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/Users/eisenham/Documents/ssbdev/jwst/jwst/stpipe/step.py:143: in from_cmdline
    return cmdline.step_from_cmdline(args)
/Users/eisenham/Documents/ssbdev/jwst/jwst/stpipe/cmdline.py:320: in step_from_cmdline
    step.run(*positional)
/Users/eisenham/Documents/ssbdev/jwst/jwst/stpipe/step.py:361: in run
    result = self.process(*args)
/Users/eisenham/Documents/ssbdev/jwst/jwst/pipeline/calwebb_spec3.py:125: in process
    pass
/Users/eisenham/Documents/ssbdev/jwst/jwst/stpipe/step.py:361: in run
    result = self.process(*args)
/Users/eisenham/Documents/ssbdev/jwst/jwst/resample/resample_spec_step.py:76: in process
    fillval=self.fillval, good_bits=self.good_bits)
/Users/eisenham/Documents/ssbdev/jwst/jwst/resample/resample_spec.py:92: in __init__
    self.build_nirspec_output_wcs()
/Users/eisenham/Documents/ssbdev/jwst/jwst/resample/resample_spec.py:246: in build_nirspec_output_wcs
    bounding_box_grid = wnew.backward_transform(ra, dec, lam)
/Users/eisenham/anaconda3/envs/spec3/lib/python3.5/site-packages/astropy/modeling/core.py:401: in __call__
    ('equivalencies', None)])
/Users/eisenham/anaconda3/envs/spec3/lib/python3.5/site-packages/astropy/modeling/core.py:382: in __call__
    return super(cls, self).__call__(*inputs, **kwargs)
/Users/eisenham/anaconda3/envs/spec3/lib/python3.5/site-packages/astropy/modeling/core.py:785: in __call__
    outputs = self.evaluate(*chain(inputs, parameters))
/Users/eisenham/anaconda3/envs/spec3/lib/python3.5/site-packages/astropy/modeling/core.py:2816: in evaluate
    return self.__class__.evaluate(*args)
/Users/eisenham/anaconda3/envs/spec3/lib/python3.5/site-packages/astropy/modeling/core.py:2285: in evaluate
    result = cls._evaluate(inputs, params)
/Users/eisenham/anaconda3/envs/spec3/lib/python3.5/site-packages/astropy/modeling/core.py:2109: in <lambda>
    return (lambda inputs, params: g[0](f[0](inputs, params), params),
/Users/eisenham/anaconda3/envs/spec3/lib/python3.5/site-packages/astropy/modeling/core.py:2120: in <lambda>
    return (lambda inputs, params: (f[0](inputs[:f[1]], params) +
/Users/eisenham/anaconda3/envs/spec3/lib/python3.5/site-packages/astropy/modeling/core.py:2109: in <lambda>
    return (lambda inputs, params: g[0](f[0](inputs, params), params),
/Users/eisenham/anaconda3/envs/spec3/lib/python3.5/site-packages/astropy/modeling/core.py:2731: in <lambda>
    evaluate(*chain(inputs, islice(params, n_params)))
/Users/eisenham/anaconda3/envs/spec3/lib/python3.5/site-packages/astropy/modeling/rotations.py:240: in evaluate
    phi = (np.pi / 2 + lon)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <Quantity [ 0.00189193] deg>, obj = array([  2.31584178e+77]), context = (<ufunc 'add'>, (1.5707963267948966, <Quantity [ 0.00189193] deg>), 0)

    def __array_prepare__(self, obj, context=None):
        # This method gets called by Numpy whenever a ufunc is called on the
        # array. The object passed in ``obj`` is an empty version of the
        # output array which we can e.g. change to an array sub-class, add
        # attributes to, etc. After this is called, then the ufunc is called
        # and the values in this empty array are set.

        # If no context is set, just return the input
        if context is None:
            return obj

        # Find out which ufunc is being used
        function = context[0]

        from .quantity_helper import UNSUPPORTED_UFUNCS, UFUNC_HELPERS

        # Check whether we even support this ufunc
        if function in UNSUPPORTED_UFUNCS:
            raise TypeError("Cannot use function '{0}' with quantities"
                            .format(function.__name__))

        # Now find out what arguments were passed to the ufunc, usually, this
        # will include at least the present object, and another, which could
        # be a Quantity, or a Numpy array, etc. when using two-argument ufuncs.
        args = context[1][:function.nin]
        units = [getattr(arg, 'unit', None) for arg in args]

        # If the ufunc is supported, then we call a helper function (defined
        # in quantity_helper.py) which returns the scale by which the inputs
        # should be multiplied before being passed to the ufunc, as well as
        # the unit the output from the ufunc will have.
        if function in UFUNC_HELPERS:
            converters, result_unit = UFUNC_HELPERS[function](function, *units)
            if function.nout > 1:
                result_unit = result_unit[context[2]]
        else:
            raise TypeError("Unknown ufunc {0}.  Please raise issue on "
                            "https://github.com/astropy/astropy"
                            .format(function.__name__))

        if any(converter is False for converter in converters):
            # for two-argument ufuncs with a quantity and a non-quantity,
            # the quantity normally needs to be dimensionless, *except*
            # if the non-quantity can have arbitrary unit, i.e., when it
            # is all zero, infinity or NaN.  In that case, the non-quantity
            # can just have the unit of the quantity
            # (this allows, e.g., `q > 0.` independent of unit)
            maybe_arbitrary_arg = args[converters.index(False)]
            try:
                if _can_have_arbitrary_unit(maybe_arbitrary_arg):
                    converters = [None, None]
                else:
                    raise UnitsError("Can only apply '{0}' function to "
                                     "dimensionless quantities when other "
                                     "argument is not a quantity (unless the "
                                     "latter is all zero/infinity/nan)"
>                                    .format(function.__name__))
E                                    astropy.units.core.UnitsError: Can only apply 'add' function to dimensionless quantities when other argument is not a quantity (unless the latter is all zero/infinity/nan)

/Users/eisenham/anaconda3/envs/spec3/lib/python3.5/site-packages/astropy/units/quantity.py:468: UnitsError
----------------------------------------------------------------------------- Captured stdout call -----------------------------------------------------------------------------
----------------------------------------------------------------------
ERROR RUNNING STEP 'Spec3Pipeline':
    Can only apply 'add' function to dimensionless quantities when
    other argument is not a quantity (unless the latter is all
    zero/infinity/nan)
Traceback (most recent call last):
  File "/Users/eisenham/Documents/ssbdev/jwst/jwst/stpipe/cmdline.py", line 320, in step_from_cmdline
    step.run(*positional)
  File "/Users/eisenham/Documents/ssbdev/jwst/jwst/stpipe/step.py", line 361, in run
    result = self.process(*args)
  File "/Users/eisenham/Documents/ssbdev/jwst/jwst/pipeline/calwebb_spec3.py", line 125, in process
    pass
  File "/Users/eisenham/Documents/ssbdev/jwst/jwst/stpipe/step.py", line 361, in run
    result = self.process(*args)
  File "/Users/eisenham/Documents/ssbdev/jwst/jwst/resample/resample_spec_step.py", line 76, in process
    fillval=self.fillval, good_bits=self.good_bits)
  File "/Users/eisenham/Documents/ssbdev/jwst/jwst/resample/resample_spec.py", line 92, in __init__
    self.build_nirspec_output_wcs()
  File "/Users/eisenham/Documents/ssbdev/jwst/jwst/resample/resample_spec.py", line 246, in build_nirspec_output_wcs
    bounding_box_grid = wnew.backward_transform(ra, dec, lam)
  File "/Users/eisenham/anaconda3/envs/spec3/lib/python3.5/site-packages/astropy/modeling/core.py", line 401, in __call__
    ('equivalencies', None)])
  File "/Users/eisenham/anaconda3/envs/spec3/lib/python3.5/site-packages/astropy/modeling/core.py", line 382, in __call__
    return super(cls, self).__call__(*inputs, **kwargs)
  File "/Users/eisenham/anaconda3/envs/spec3/lib/python3.5/site-packages/astropy/modeling/core.py", line 785, in __call__
    outputs = self.evaluate(*chain(inputs, parameters))
  File "/Users/eisenham/anaconda3/envs/spec3/lib/python3.5/site-packages/astropy/modeling/core.py", line 2816, in evaluate
    return self.__class__.evaluate(*args)
  File "/Users/eisenham/anaconda3/envs/spec3/lib/python3.5/site-packages/astropy/modeling/core.py", line 2285, in evaluate
    result = cls._evaluate(inputs, params)
  File "/Users/eisenham/anaconda3/envs/spec3/lib/python3.5/site-packages/astropy/modeling/core.py", line 2109, in <lambda>
    return (lambda inputs, params: g[0](f[0](inputs, params), params),
  File "/Users/eisenham/anaconda3/envs/spec3/lib/python3.5/site-packages/astropy/modeling/core.py", line 2120, in <lambda>
    return (lambda inputs, params: (f[0](inputs[:f[1]], params) +
  File "/Users/eisenham/anaconda3/envs/spec3/lib/python3.5/site-packages/astropy/modeling/core.py", line 2109, in <lambda>
    return (lambda inputs, params: g[0](f[0](inputs, params), params),
  File "/Users/eisenham/anaconda3/envs/spec3/lib/python3.5/site-packages/astropy/modeling/core.py", line 2731, in <lambda>
    evaluate(*chain(inputs, islice(params, n_params)))
  File "/Users/eisenham/anaconda3/envs/spec3/lib/python3.5/site-packages/astropy/modeling/rotations.py", line 240, in evaluate
    phi = (np.pi / 2 + lon)
  File "/Users/eisenham/anaconda3/envs/spec3/lib/python3.5/site-packages/astropy/units/quantity.py", line 468, in __array_prepare__
    .format(function.__name__))
astropy.units.core.UnitsError: Can only apply 'add' function to dimensionless quantities when other argument is not a quantity (unless the latter is all zero/infinity/nan)

----------------------------------------------------------------------
----------------------------------------------------------------------------- Captured stderr call -----------------------------------------------------------------------------
2017-06-26 22:58:32,068 - stpipe.Spec3Pipeline - INFO - Spec3Pipeline instance created.
2017-06-26 22:58:32,070 - stpipe.Spec3Pipeline.cube_build - INFO - CubeBuildStep instance created.
2017-06-26 22:58:32,071 - stpipe.Spec3Pipeline.resample_spec - INFO - ResampleSpecStep instance created.
2017-06-26 22:58:32,072 - stpipe.Spec3Pipeline.outlier_detection - INFO - OutlierDetectionStep instance created.
2017-06-26 22:58:32,073 - stpipe.Spec3Pipeline.skymatch - INFO - SkyMatchStep instance created.
2017-06-26 22:58:32,074 - stpipe.Spec3Pipeline.extract_1d - INFO - Extract1dStep instance created.
2017-06-26 22:58:32,074 - stpipe - INFO - Hostname: inanna2.local
2017-06-26 22:58:32,074 - stpipe - INFO - OS: Darwin
2017-06-26 22:58:32,140 - stpipe.Spec3Pipeline - INFO - Step Spec3Pipeline running with args ('/var/folders/z7/hcykr_n546s_5033t9rj5d200000zj/T/tmpqyei77i0.json',).
2017-06-26 22:58:32,140 - stpipe.Spec3Pipeline - INFO - Starting calwebb_spec3 ...
2017-06-26 23:13:00,765 - stpipe.Spec3Pipeline - INFO - Convert from exposure-based to source-based data.
2017-06-26 23:31:28,850 - stpipe.Spec3Pipeline - INFO - Step.save_model level3_cal.fits
2017-06-26 23:31:51,917 - stpipe.Spec3Pipeline.outlier_detection - INFO - Step outlier_detection running with args (<jwst.datamodels.source_container.SourceModelContainer object at 0x1642a19b0>,).
2017-06-26 23:31:51,918 - stpipe.Spec3Pipeline.outlier_detection - INFO - Step skipped.
2017-06-26 23:31:51,920 - stpipe.Spec3Pipeline.outlier_detection - INFO - Step outlier_detection done
2017-06-26 23:32:08,460 - stpipe.Spec3Pipeline.resample_spec - INFO - Step resample_spec running with args (<jwst.datamodels.source_container.SourceModelContainer object at 0x1642a19b0>,).
------------------------------------------------------------------------------ Captured log call -------------------------------------------------------------------------------
step.py                    289 INFO     Spec3Pipeline instance created.
step.py                    289 INFO     CubeBuildStep instance created.
step.py                    289 INFO     ResampleSpecStep instance created.
step.py                    289 INFO     OutlierDetectionStep instance created.
step.py                    289 INFO     SkyMatchStep instance created.
step.py                    289 INFO     Extract1dStep instance created.
cmdline.py                 281 INFO     Hostname: inanna2.local
cmdline.py                 281 INFO     Hostname: inanna2.local
cmdline.py                 282 INFO     OS: Darwin
cmdline.py                 282 INFO     OS: Darwin
step.py                    340 INFO     Step Spec3Pipeline running with args ('/var/folders/z7/hcykr_n546s_5033t9rj5d200000zj/T/tmpqyei77i0.json',).
calwebb_spec3.py            54 INFO     Starting calwebb_spec3 ...
util.py                    145 DEBUG    Opening as <class 'jwst.datamodels.multislit.MultiSlitModel'>
util.py                    145 DEBUG    Opening as <class 'jwst.datamodels.multislit.MultiSlitModel'>
util.py                    145 DEBUG    Opening as <class 'jwst.datamodels.multislit.MultiSlitModel'>
util.py                    145 DEBUG    Opening as <class 'jwst.datamodels.multislit.MultiSlitModel'>
util.py                    145 DEBUG    Opening as <class 'jwst.datamodels.multislit.MultiSlitModel'>
util.py                    145 DEBUG    Opening as <class 'jwst.datamodels.multislit.MultiSlitModel'>
util.py                    145 DEBUG    Opening as <class 'jwst.datamodels.multislit.MultiSlitModel'>
util.py                    145 DEBUG    Opening as <class 'jwst.datamodels.multislit.MultiSlitModel'>
util.py                    145 DEBUG    Opening as <class 'jwst.datamodels.multislit.MultiSlitModel'>
util.py                    145 DEBUG    Opening as <class 'jwst.datamodels.multislit.MultiSlitModel'>
util.py                    145 DEBUG    Opening as <class 'jwst.datamodels.multislit.MultiSlitModel'>
util.py                    145 DEBUG    Opening as <class 'jwst.datamodels.multislit.MultiSlitModel'>
calwebb_spec3.py            94 INFO     Convert from exposure-based to source-based data.
step.py                    727 INFO     Step.save_model level3_cal.fits
step.py                    340 INFO     Step outlier_detection running with args (<jwst.datamodels.source_container.SourceModelContainer object at 0x1642a19b0>,).
step.py                    357 INFO     Step skipped.
step.py                    422 INFO     Step outlier_detection done
step.py                    340 INFO     Step resample_spec running with args (<jwst.datamodels.source_container.SourceModelContainer object at 0x1642a19b0>,).
resample_spec.py           175 DEBUG    Center of bounding box: 586  12
resample_spec.py           175 DEBUG    Center of bounding box: 586  12
resample_spec.py           198 DEBUG    Slit center pix: 6.0
resample_spec.py           198 DEBUG    Slit center pix: 6.0
resample_spec.py           201 DEBUG    Slit center: <SkyCoord (ICRS): (ra, dec) in deg
    ( 0.10839945,  0.01756356)>
resample_spec.py           201 DEBUG    Slit center: <SkyCoord (ICRS): (ra, dec) in deg
    ( 0.10839945,  0.01756356)>
resample_spec.py           202 DEBUG    Fiducial: (array([ 0.10838543,  0.01754757]), array([ 1.68738814]))
resample_spec.py           202 DEBUG    Fiducial: (array([ 0.10838543,  0.01754757]), array([ 1.68738814]))
resample_spec.py           204 DEBUG    Slit angular size: 1.397033282713736
resample_spec.py           204 DEBUG    Slit angular size: 1.397033282713736
resample_spec.py           207 DEBUG    Offset up the slit: (<Angle 0.921018333272347 arcsec>, <Angle 1.0504414418678942 arcsec>)
resample_spec.py           207 DEBUG    Offset up the slit: (<Angle 0.921018333272347 arcsec>, <Angle 1.0504414418678942 arcsec>)
resample_spec.py           214 DEBUG    Spatial scale: 0.10733564787219323
resample_spec.py           214 DEBUG    Spatial scale: 0.10733564787219323
resample_spec.py           220 DEBUG    Slit rotation angle: 2.8016740138681184
resample_spec.py           220 DEBUG    Slit rotation angle: 2.8016740138681184
========================================================================= 1 failed in 2084.39 seconds ==========================================================================
```

**Comment by stscieisenhamer** [comment]
In trying to reproduce this issue, this error no longer occurs. But of course, there are different errors which will be documented in other issues.

**Comment by hbushouse** [comment]
resample_spec for NIRSpec MSA data was working in the B7.1 delivery, so this was likely fixed by PR's such as #1248 and #1412.
